src/Concat.java

```diff
@@ -0,0 +1,132 @@
+import java.util.Collections;
+import java.util.concurrent.ExecutionException;
+
+public class Concat {
+
+    public List<String> readWordsFromFile(String fileName) {
+        List<String> resultList = new LinkedList<>();
+        try {
+            File file = new File(fileName);
+            FileReader fr = new FileReader(file);
+            BufferedReader reader = new BufferedReader(fr);
+            String line = reader.readLine();
+
+            while (line != null) {
+                List<String> words = Arrays.asList(line.split("\\s+"));
+                resultList.addAll(words);
+                line = reader.readLine();
+            }
+
+        } catch (FileNotFoundException e) {
+            throw new RuntimeException("FileNotFoundException has happened with such message : " + e.getMessage());
+        } catch (IOException e) {
+            throw new RuntimeException("IOException has happened with such message : " + e.getMessage());
+        }
+        Collections.sort(resultList);
+
+        return resultList;
+    }
+
+    public List<String> findConcatenatedWords(List<String> words) {
+        List<String> concatenatedWords = new LinkedList<>();
+        ExecutorService executor = Executors.newFixedThreadPool(Runtime.getRuntime().availableProcessors());
+        List<Future<List<String>>> futureLinkedList = new LinkedList<>();
+
+        int partSize = 100;
+        int prev = 0;
+        int next = partSize;
+        boolean exit = false;
+
+        do {
+            if (words.size() - next < 0) {
+                next = words.size();
+            }
+            List<String> part = words.subList(prev, next);
+            if (!part.isEmpty()) {
+                Callable<List<String>> callable = new ConcatCallable(part, words);
+                Future<List<String>> future = executor.submit(callable);
+                futureLinkedList.add(future);
+                prev = next;
+                next = prev + partSize;
+            } else {
+                exit = true;
+            }
+        } while (!exit);
+
+        for (Future<List<String>> listFuture : futureLinkedList) {
+            try {
+                concatenatedWords.addAll(listFuture.get());
+            } catch (ExecutionException e) {
+                throw new RuntimeException("ExecutionException has happened with such message : " + e.getMessage());
+            } catch (InterruptedException e) {
+                throw new RuntimeException("InterruptedException has happened with such message : " + e.getMessage());
+            }
+        }
+
+        executor.shutdown();
+        return concatenatedWords;
+    }
+
+    public List<String> findTheLongestConcatenatedWord(List<String> words) {
+        List<String> concatenatedWords = findConcatenatedWords(words);
+        List<Integer> lengths = getLengths(concatenatedWords);
+
+        if (lengths.size() > 0) {
+            return getWordsWithLength(concatenatedWords, lengths, 1);
+        } else {
```

Comment:
```suggestion
        }
```
якщо стоїть return, то не потрібно else, оскільки метод закінчує роботу після виконання return